paasta_tools/marathon_tools.py

```diff
@@ -204,33 +203,41 @@ def get_instances(self):
         :returns: The number of instances specified in the config, 0 if not
                   specified or if desired_state is not 'start'."""
         if self.get_desired_state() == 'start':
-            instances = self.get_min_instances()
             max_instances = self.get_max_instances()
             if max_instances is not None:
+                instances = self.get_min_instances()
                 try:
                     zk_instances = get_instances_from_zookeeper(
                         service=self.service,
                         instance=self.instance,
                     )
                 except NoNodeError:  # zookeeper doesn't have instance data for this app
-                    pass
+                    return instances
                 else:
-                    instances = max(
+                    return max(
                         min(zk_instances, max_instances),
```

Comment:
but won't zk_instances be not set by the time we get here if we get a NoNodeError ?